### PR TITLE
Changing from '.directory' blob names to directory-name-as-blob-name.…

### PR DESCRIPTION
**Pull request by asorrin-msft**
…  Should be back-compat.

### CODE REVIEW

File: blobfuse/utilities.cpp [hunk 1]
```diff
@@ -382,6 +415,22 @@ int azs_getattr(const char *path, struct stat *stbuf)
...
     if ((errno == 0) && blob_property.valid())
     {
+        if (is_directory_blob(blob_property.size, blob_property.metadata))
+        {
+            if (AZS_PRINT)
+            {
+                fprintf(stdout, "Directory found!  Name = %s\n", path);
+            }
+            stbuf->st_mode = S_IFDIR | default_permission;
+            // If st_nlink = 2, means direcotry is empty.
```
Comment (AlexanderYukhanov) [inline blobfuse/utilities.cpp]: misprint

Comment (asorrin-msft) [reply blobfuse/utilities.cpp]: fixed

File: azure-storage-cpp-lite/src/blob/blob_client.cpp [hunk 2]
```diff
@@ -205,6 +205,7 @@ std::future<storage_outcome<list_blobs_hierarchical_response>> blob_client::list
 
     auto request = std::make_shared<list_blobs_hierarchical_request>(container, delimiter, continuation_token, prefix);
     request->set_maxresults(10000);
+    request->set_includes(list_blobs_hierarchical_request_base::include::metadata);
```
[the remark this answers is not in the document]
Comment (asorrin-msft) [reply azure-storage-cpp-lite/src/blob/blob_client.cpp]: fixed

File: azure-storage-cpp-lite/src/blob/blob_client.cpp [hunk 3]
```diff
@@ -245,9 +246,9 @@ storage_outcome<blob_property> blob_client::get_blob_property(const std::string
         auto& headers = http->get_headers();
         for (auto iter = headers.begin(); iter != headers.end(); ++iter)
         {
-            if (iter->first.find("x-ms-metadata-") == 0)
+            if (iter->first.find("x-ms-meta-") == 0)
             {
-                blobProperty.metadata.push_back(std::make_pair(iter->first, iter->second));
+                blobProperty.metadata.push_back(std::make_pair(iter->first.substr(10), iter->second.substr(0, iter->second.size() - 2)));
```
[the remark this answers is not in the document]
Comment (asorrin-msft) [reply azure-storage-cpp-lite/src/blob/blob_client.cpp]: fixed

**Pull request merged**